### PR TITLE
#216 update user area with current user

### LINKED ISSUE #216: Update the userArea to login with another user
Update the UserArea to use the information from user updated on the User context. And update the code to allow login with a different user (after logout the current one!).

Ideally make the user area more similar to what is in the UI design. Don't spend too long on this layout change! This should be just a tiny layout refactor to look a little bit better. 
Actually making the User Area look exactly like designed in **Epic#10 - Layout and menus** #148.

Design:
Screen Shot 2020-12-14 at 10 51 00 AM

### PR DESCRIPTION
**Pull request by nmadruga**
Fixes #216 

### Changes
- Layout on UserArea to look more similar to UI Designs
- Add selection to change current user on the go
- Move hook `useGetUserInfo` to a utility function, it receives the dispatch method to set a new user after a successful fetch
- [EDITED] Move `login` and `logout` function to be exported from the **UserState** context. It also checks on load if there is a persisted user (in local storage) to login the user.

### Changes unrelated
- Update query to get Current user information to also store the `id` and uses the id on the existing Reviewer query.
- Update a few hardcoded strings to be using a constant or message define in the specific files

### CODE REVIEW

File: src/containers/User/Login.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ const Login: React.FC = () => {
 
 export default Login
 
-function hashPassword(password: string) {
+export function hashPassword(password: string) {
```
Comment (CarlosNZ) [inline src/containers/User/Login.tsx:76]: I wouldn't bother with this, since password hashing will be done on the Server, as per #149.

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Just before I look at this any deeper:
- There's a problem been introduced with the order things are loading, I think. When I log in, the Home page (Home.tsx) doesn't show the first name in "Hello XXX, welcome to the Dashboard"., and the name doesn't show in the Users dropdown.  If I switch users, it shows the new user correctly, but when I reload the page, the usernames aren't showing again.

File: src/containers/User/Login.tsx (hunk 1 above)
Comment (nmadruga) [reply src/containers/User/Login.tsx:77]: Ok, I'll remove that

**Comment by nmadruga** [comment]
Cool. It took me sometime on this one, but I think it's good to be re-reviewing now.

Thanks for spotting this problem, Carl.

File: src/containers/Main/AppWrapper.tsx [hunk 2]
```diff
@@ -30,15 +30,21 @@ const AppWrapper: React.FC = () => {
...
     return (
       <>
         <UserArea />
         <SiteLayout />
         <Footer />
       </>
     )
-  else return <Redirect to={{ pathname: '/login', state: { from: location.pathname } }} />
+  } else {
+    console.log('NOT LOGGED')
```
Comment (CarlosNZ) [inline src/containers/Main/AppWrapper.tsx:44]: Delete log.

**Review by CarlosNZ — APPROVED** [review]
Nice, working really well now and all seems very logical and clean. Thx.

2 console.logs to remove.

File: src/containers/Main/AppWrapper.tsx [hunk 3]
```diff
@@ -30,15 +30,21 @@ const AppWrapper: React.FC = () => {
 }
 
 const AuthenticatedContent: React.FC = () => {
-  if (isLoggedIn())
+  if (isLoggedIn()) {
+    console.log('LOGGED')
+
```
[the remark this answers is not in the document]
Comment (nmadruga) [reply src/containers/Main/AppWrapper.tsx:35]: Thanks! Removing it now

**Pull request merged**